packages/@storefront/structure/src/generic-paging/index.ts

```diff
@@ -0,0 +1,152 @@
+import { provide, tag, Tag } from '@storefront/core';
+
+@provide('paging')
+@tag('gb-generic-paging', require('./index.html'))
+class GenericPaging {
+  props: GenericPaging.Props = {
+    showIcons: true,
+    showLabels: true,
+    numericLabels: false,
+    labels: { first: 'First', next: 'Next', prev: 'Prev', last: 'Last' },
+    icons: {},
+    pageSize: 20,
+    current: 1,
+    itemCount: 0,
+    limit: 5,
+  };
+
+  state: GenericPaging.State = {
+    range: [],
+    backDisabled: true,
+    forwardDisabled: true,
+    lowOverflow: false,
+    highOverflow: false,
+    firstPage: () => {
+      if (typeof this.props.onFirstPage === 'function') {
+        this.props.onFirstPage();
+      }
+    },
+    lastPage: () => {
+      if (typeof this.props.onLastPage === 'function') {
+        this.props.onLastPage();
+      }
+    },
+    prevPage: () => {
+      if (typeof this.props.onPrevPage === 'function') {
+        this.props.onPrevPage();
+      }
+    },
+    nextPage: () => {
+      if (typeof this.props.onNextPage === 'function') {
+        this.props.onNextPage();
+      }
+    },
+    switchPage: (page: number) => () => {
```

Comment:
@JoeyDeol Why does this need to be a higher order function?

Reply:
The outer function is called by Riot when generating the list items to create a closure for the actual click handler. This also preserves the original interaface of the old `$paging` alias.

Reply:
Noted. Resolving.